services/paginateResponse.js

```diff
@@ -0,0 +1,153 @@
+const proverb = require('../models/proverb')
+
+function addNextPreviousTotalObj (page, limit, arrLength) {
+  const startIndex = (page - 1) * limit
+  const endIndex = page * limit
+
+  const results = {}
+
+  results.total_pages = Math.ceil(arrLength / limit)
+
+  if (endIndex < arrLength) {
+    results.next = {
+      page: page + 1,
+      limit: limit
+    }
+  }
+
+  if (startIndex > 0) {
+    results.previous = {
+      page: page - 1,
+      limit: limit
+    }
+  }
+  return { results, startIndex, endIndex }
+}
+function sortArr (arr, direction) {
+  return arr.sort((a, b) => {
+    if (direction === 'asc') {
+      return a.date - b.date
+    } else {
+      return b.date - a.date
+    }
+  })
+}
+
+function paginateResponse (model) {
+  return async (req, res, next) => {
+    const page = parseInt(req.query.page)
+    const limit = parseInt(req.query.limit)
+    const sort = req.query.sort
+    const approved = req.query.approved
+
+    const { results, startIndex } = addNextPreviousTotalObj(
+      page,
+      limit,
+      await model.countDocuments().exec()
+    )
+
+    if (approved === undefined) {
+      try {
+        results.total_pages = Math.ceil(
+          (await model
+            .find({})
+            .countDocuments()
+            .exec()) / limit
+        )
+
+        results.results = await model
+          .find({})
+          .sort({ date: sort ? `${sort}` : null })
+          .limit(limit)
+          .skip(startIndex)
+          .exec()
+        res.paginatedResults = results
+        next()
+      } catch (error) {
+        res.status(500).json({ msg: error.message })
+      }
+      return
```

Comment:
```suggestion
        return next()
      } catch (error) {
        return res.status(500).json({ msg: error.message })
      }
```

Here another approach would be to return inside the try catch statement